src/Aspire.Hosting/ApplicationModel/ParameterResource.cs

```diff
@@ -63,5 +64,15 @@ public string Value
     /// </summary>
     public string ValueExpression => $"{{{Name}.value}}";
 
+    /// <summary>
+    /// The configuration key for this parameter. The default format is "ConnectionStrings:{Name}" if the parameter is a connection string,
+    /// otherwise it is "Parameters:{Name}".
+    /// </summary>
+    internal string ConfigurationKey
+    {
+        get => _configurationKey ?? (IsConnectionString ? $"ConnectionStrings:{Name}" : $"Parameters:{Name}");
+        set => _configurationKey = value;
+    }
```

Comment:
Glad this is internal. I want to avoid strictly tying the concept of a parameter/connection string to the .NET configuration store. They are connected for convienence however.